var/spack/repos/builtin/packages/py-torch-cluster/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyTorchCluster(PythonPackage):
+    """This package consists of a small extension library of
+    highly optimized graph cluster algorithms for the use in
+    PyTorch."""
+
+    homepage = "https://github.com/rusty1s/pytorch_cluster"
+    url      = "https://github.com/rusty1s/pytorch_cluster/archive/1.5.7.tar.gz"
+
+    version('1.5.7', sha256='71701d2f7f3e458ebe5904c982951349fdb60e6f1654e19c7e102a226e2de72e')
+
+    variant('cuda', default=False)
+
+    depends_on('python@3.6:', type=('build', 'run'))
```

Comment:
Missing build dep on setuptools and possibly pytest-runner

Reply:
I keep assuming `setuptools` is implied by `PythonPackage`.

If it isn't, should it be?

Reply:
It isn't, and it shouldn't be.

`setuptools` is an optional tool that packages can use to build from source. Not all packages use it, although we're close to 90% of Spack Python packages using it.